examples/authentication-swapping/docker-compose.yaml

```diff
@@ -0,0 +1,149 @@
+ # For simplicity of local development, we are using port based routing, disabled encryption from client to gateway and gateway to broker.
+ # In a production environment, you should use TLS encryption from client to gateway and gateway to broker. And use SNI routing for routing requests to the correct broker.
+
+  gateway:
+    image: "${GATEWAY_IMAGE}"
+    container_name: gateway
+    depends_on:
+      - kafka-1
+      - vault
+    volumes:
+      - ${GATEWAY_JAAS_CONF_FOR_GW_AUTHN}:/etc/gateway/config/jaas-config-for-gw-authn.conf # loading JAAS config for SASL authentication at Gateway
+      - ${GATEWAY_JAAS_TEMPLATE_FOR_GW_SWAPPING}:/etc/gateway/config/jaas-template-for-gw-swapping.conf # loading JAAS template for SASL authentication at Gateway
+    environment:
+      GATEWAY_CONFIG: | 
+        gateway:
+          admin:
+            endpoints:
+              metrics: true
+          secretStores:
+            - name: vault-secret-store
+              provider:
+                type: Vault
+                config:
+                  address: http://vault:8200
+                  authToken: vault-plaintext-root-token
+                  path: secret/testing
+                  separator: /
+      
+          streamingDomains:
+            - name: sample-domain  
+              type: kafka
+              kafkaCluster:
+                name: kafka-cluster-1 
+                nodeIdRanges:
+                  - name: d
+                    start: 1 
+                    end: 5
+                bootstrapServers:
+                  - id: internal-kafka-listener  
+                    endpoint: "kafka-1:44444" 
+          routes:
+            - name: auth-swap-route
+              endpoint: "host.docker.internal:19092" 
+              brokerIdentificationStrategy:
+                type: port 
+              streamingDomain:
+                name: sample-domain
+                bootstrapServerId: internal-kafka-listener 
+              security:
+                auth: swap 
+                swapConfig:
+                  clientAuth:
+                    sasl:
+                      mechanism: PLAIN
+                      callbackHandlerClass: org.apache.kafka.common.security.plain.internals.PlainServerCallbackHandler
+                      jaasConfig:
+                        file: /etc/gateway/config/jaas-config-for-gw-authn.conf
+                  secretStore: vault-secret-store
+                  clusterAuth:
+                    sasl:
+                      mechanism: PLAIN
+                      callbackHandlerClass: org.apache.kafka.common.security.authenticator.SaslClientCallbackHandler
+                      jaasConfig:
+                        file: /etc/gateway/config/jaas-template-for-gw-swapping.conf
+
+
+    ports: # mapping gateway ports to the host ports. These are the ports on which the gateway will listen for requests.
+      - "19092:19092"
```

Comment:
nodeIdRanges are 1 to 5. ports from 19092 (bootstrap), 19093 (nodeId=1), 19094, 19095, 19096, 19097 (nodeId=5) should be exposed.